exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -138,7 +170,137 @@ public static TestKit forService(Class<? extends ServiceModule> serviceModule) {
+    byte[] block = nativeCreateBlockWithTransactions(nativeHandle.get(), transactionMessagesArr);
+    return BLOCK_SERIALIZER.fromBytes(block);
+  }
+
+  /**
+   * Creates a block with all in-pool transactions. Transactions are applied in the lexicographical
+   * order of their hashes.
+   *
+   * @return created block
+   */
+  public Block createBlock() {
+    List<TransactionMessage> inPoolTransactions =
+        findTransactionsInPool(transactionMessage -> true);
+    checkTransactions(inPoolTransactions);
+    byte[] block = nativeCreateBlock(nativeHandle.get());
+    return BLOCK_SERIALIZER.fromBytes(block);
+  }
+
+  private void checkTransactions(List<TransactionMessage> transactionMessages) {
+    for (TransactionMessage transactionMessage: transactionMessages) {
+      checkTransaction(transactionMessage);
+    }
+  }
+
+  private void checkTransaction(TransactionMessage transactionMessage) {
+    short serviceId = transactionMessage.getServiceId();
+    RawTransaction rawTransaction = toRawTransaction(transactionMessage);
+    if (!services.containsKey(serviceId)) {
+      String message = String.format("Unknown service id (%s) in transaction (%s)",
+          serviceId, rawTransaction);
+      throw new IllegalArgumentException(message);
+    }
+    Service service = services.get(serviceId);
+    try {
+      service.convertToTransaction(rawTransaction);
+    } catch (Throwable conversionError) {
+      String message = String.format("Service (%s) with id=%s failed to convert transaction (%s)."
+          + " Make sure that the submitted transaction is correctly serialized, and the service's"
+          + " TransactionConverter implementation is correct and handles this transaction as"
+          + " expected.", service.getName(), serviceId, rawTransaction);
+      throw new IllegalArgumentException(message, conversionError);
+    }
+  }
+
+  @VisibleForTesting
+  static RawTransaction toRawTransaction(TransactionMessage transactionMessage) {
+    return RawTransaction.newBuilder()
+        .serviceId(transactionMessage.getServiceId())
+        .transactionId(transactionMessage.getTransactionId())
+        .payload(transactionMessage.getPayload())
+        .build();
+  }
+
+  /**
+   * Returns a list of in-pool transactions that match the given predicate.
+   */
+  public List<TransactionMessage> findTransactionsInPool(Predicate<TransactionMessage> predicate) {
+    return withSnapshot((view) -> {
+      Blockchain blockchain = Blockchain.newInstance(view);
+      MapIndex<HashCode, TransactionMessage> txMessages = blockchain.getTxMessages();
+      KeySetIndexProxy<HashCode> poolTxsHashes = blockchain.getTransactionPool();
+      Set<HashCode> poolTxsHashesSet = toSet(poolTxsHashes);
+      return poolTxsHashesSet.stream()
+          .map(txMessages::get)
+          .collect(toList());
+    });
+  }
+
+  private <T> Set<T> toSet(KeySetIndexProxy<T> setIndex) {
+    return Sets.newHashSet(setIndex.iterator());
```

Comment:
I would suggest using LinkedHashSet to preserve iteration order (= the order in which txs will be committed).
```suggestion
    return Sets.newLinkedHashSet(setIndex.iterator());
```

Reply:
Or, better — just stream them directly, with no intermediate copy — I'll push that directly